packages/admin/admin/src/common/readOnlyField/ReadOnlyTextField.tsx

```diff
@@ -0,0 +1,64 @@
+import { Lock } from "@comet/admin-icons";
+import { css } from "@mui/material/styles";
+import { FunctionComponent, ReactNode } from "react";
+
+import { createComponentSlot } from "../../helpers/createComponentSlot";
+import { ReadOnlyFieldLabel } from "./ReadOnlyFieldLabel";
+
+export const ReadOnlyTextField: FunctionComponent<{
+    label?: ReactNode;
+    value?: string | null;
+    className?: string;
+}> = ({ label, value, className }) => (
+    <Wrapper className={className}>
+        {label && <ReadOnlyFieldLabel label={label} />}
+        {value && (
+            <Box>
+                <TextContent>{value}</TextContent>
+                <Lock />
+            </Box>
+        )}
+    </Wrapper>
```

Comment:
Most for the readonly fields are pretty similar. What do you think about creating a base component that supports label and adornments that is then used by the components?

Reply:
you could also use FieldContainer